wiki-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java

```diff
@@ -0,0 +1,181 @@
+import org.exoplatform.wiki.service.image.ResizeImageService;
+import org.exoplatform.wiki.service.impl.BeanToJsons;
+import org.exoplatform.wiki.service.impl.DraftData;
+import org.exoplatform.wiki.service.impl.SpaceBean;
+import org.exoplatform.wiki.service.related.JsonRelatedData;
+import org.exoplatform.wiki.service.related.RelatedUtil;
+import org.exoplatform.wiki.service.search.SearchResult;
+import org.exoplatform.wiki.service.search.SearchResultType;
+import org.exoplatform.wiki.service.search.TitleSearchResult;
+import org.exoplatform.wiki.service.search.WikiSearchData;
+import org.exoplatform.wiki.tree.JsonNodeData;
+import org.exoplatform.wiki.tree.TreeNode;
+import org.exoplatform.wiki.tree.TreeNode.TREETYPE;
+import org.exoplatform.wiki.tree.WikiTreeNode;
+import org.exoplatform.wiki.tree.utils.TreeUtils;
+import org.exoplatform.wiki.utils.Utils;
+import org.exoplatform.wiki.utils.WikiConstants;
+import org.exoplatform.wiki.utils.WikiHTMLSanitizer;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import javax.ws.rs.core.Response.Status;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.*;
+import java.lang.Object;
+import java.lang.Class;
+
+/**
+ * Notes REST service
+ */
+@SuppressWarnings("deprecation")
+@Path("/notes")
+public class NotesRestService implements ResourceContainer {
+
+  private final WikiService           wikiService;
+
+  private final ResourceBundleService resourceBundleService;
+
+  private static Log                  log = ExoLogger.getLogger("wiki:WikiRestService");
+
+  private final CacheControl          cc;
+
+  public NotesRestService(WikiService wikiService, ResourceBundleService resourceBundleService) {
+    this.wikiService = wikiService;
+    this.resourceBundleService = resourceBundleService;
+    cc = new CacheControl();
+    cc.setNoCache(true);
+    cc.setNoStore(true);
+  }
+
+  /**
+   * Return the note page content as html or wiki syntax.
+   * 
+   * @param text contain the data as html
+   * @return the instance of javax.ws.rs.core.Response
+   * @LevelAPI Experimental
+   */
+  @POST
+  @Path("/content/")
+  public Response getNotePageContent(@FormParam("text") String text) {
+    try {
+      String outputText = WikiHTMLSanitizer.markupSanitize(text);
+
+      return Response.ok(outputText, MediaType.TEXT_HTML).cacheControl(cc).build();
+    } catch (Exception e) {
+      log.error("Error while converting wiki page content: " + e.getMessage(), e);
+      return Response.serverError().entity(e.getMessage()).cacheControl(cc).build();
+    }
+  }
+
+  @GET
+  @Path("/{wikiType}/{wikiOwner:.+}/pages/{pageId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  public Response getPage(@Context UriInfo uriInfo,
```

Comment:
in UI, we no longer call them pages, but just ‘notes’. Wikies are now called notebooks.